contracts/market-wrapper/NounsBuilderMarkerWrapper.sol

```diff
@@ -0,0 +1,132 @@
+
+    constructor(address _auctionHouse) {
+        market = INounsBuilderAuctionHouse(_auctionHouse);
+        reservePrice = market.reservePrice();
+        minBidIncrement = market.minBidIncrement();
+    }
+
+    // ======== Private Functions =========
+
+    /**
+     * @notice Retrieve the state for the current auction
+     */
+    function _getAuctionState() private view returns (AuctionState memory) {
+        (
+            uint256 tokenId,
+            uint256 highestBid,
+            address highestBidder,
+            uint40 startTime,
+            uint40 endTime,
+            bool settled
+        ) = market.auction();
+        return AuctionState(tokenId, highestBid, highestBidder, startTime, endTime, settled);
+    }
+
+    // ======== External Functions =========
+
+    /**
+     * @notice Determine whether the current auction is valid
+     * Parameters auctionId and nftContract are unused because there's at most one ongoing auction for Nouns Builder DAOs
+     * @return TRUE if the tokenId matches the current token for sale, and the auction is neither settled nor paused
+     */
+    function auctionIdMatchesToken(
+        uint256, // auctionId
+        address, // nftContract
+        uint256 tokenId
+    ) public view override returns (bool) {
+        AuctionState memory state = _getAuctionState();
+        return (
+            state.tokenId == tokenId
+            && !state.settled
+            && !market.paused() 
+        );
+    }
+
+    /**
+     * @notice Calculate the minimum next bid for this auction
+     * Parameter auctionId is unused because there's at most one ongoing auction for Nouns Builder DAOs
+     * @return minimum bid amount
+     */
+    function getMinimumBid(uint256) external view override returns (uint256) {
+        AuctionState memory state = _getAuctionState();
+        if (state.highestBidder == address(0)) {
+            // if there are NO bids, the minimum bid is the reserve price
+            return reservePrice;
+        } else {
+            // if there ARE bids, the minimum bid is the current bid plus the increment buffer
+            return state.highestBid + (state.highestBid * minBidIncrement / 100);
+        }
+    }
+
+    /**
+     * @notice Query the current highest bidder for this auction
+     * Parameter auctionId is unused because there's at most one ongoing auction for Nouns Builder DAOs
+     * @return highest bidder
+     */
+    function getCurrentHighestBidder(uint256) external view override returns (address) {
+        AuctionState memory state = _getAuctionState();
+        return state.highestBidder;
+    }
+
+    /**
+     * @notice Submit bid to Market contract
+     * Parameter auctionId is unused because there's at most one ongoing auction for Nouns Builder DAOs
+     */
+    function bid(uint256, uint256 bidAmount) external override {
+        AuctionState memory state = _getAuctionState();
+        market.createBid{ value: bidAmount }(state.tokenId);
+    }
```

Comment:
Should this be rewritten with the ABI?
```suggestion
    function bid(uint256, uint256 bidAmount) external override {
        AuctionState memory state = _getAuctionState();
        (bool success, bytes memory returnData) = address(market).call{ value: bidAmount }(
            abi.encodeWithSignature("createBid(uint256)", state.tokenId)
        );
        require(success, string(returnData));
    }
```

Reply:
I noticed some of the other market wrappers do that.
https://github.com/PartyDAO/party-protocol/blob/2e5bc3856d4346743e379f9fefcc27a1d9de6c48/contracts/market-wrapper/ZoraMarketWrapper.sol#L91-L97
https://github.com/PartyDAO/party-protocol/blob/2e5bc3856d4346743e379f9fefcc27a1d9de6c48/contracts/market-wrapper/FoundationMarketWrapper.sol#L76-L82